pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -490,6 +491,28 @@ monitorLoop:
 	}
 }
 
+func (l *LibvirtDomainManager) CancelVMIMigration(vmi *v1.VirtualMachineInstance) error {
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Warning("failed to cancel migration, domain not found ")
+		return err
+	}
+	stats, err := dom.GetJobInfo()
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("failed to get domain job info")
+		return err
+	}
+	if stats.Type == libvirt.DOMAIN_JOB_UNBOUNDED {
+		err := dom.AbortJob()
```

Comment:
Is  this a blocking call? It might be necessary to do this in a go-routine and return fast. We are not allowed to block the controller loop in virt-handler for a long time.

Reply:
Yes, you're right. I've reworked it.